src/main/scala/barstools/tapeout/transforms/GenerateTopAndHarness.scala

```diff
@@ -7,118 +7,78 @@ import firrtl.ir._
     case _                                => None
   }.toList
 
-  lazy val rootCircuitTarget = CircuitTarget(harnessTop.get)
-
-  val topAnnos = synTop.map(st => ReParentCircuitAnnotation(rootCircuitTarget.module(st))) ++
-    topDotfOut.map(BlackBoxResourceFileNameAnno)
-
   // Dump firrtl and annotation files
   protected def dump(
     circuit:     Circuit,
-    annotations: AnnotationSeq,
-    firFile:     Option[String],
-    annoFile:    Option[String]
+    annotations: AnnotationSeq
   ): Unit = {
-    firFile.foreach { firPath =>
-      val outputFile = new java.io.PrintWriter(firPath)
-      outputFile.write(circuit.serialize)
-      outputFile.close()
-    }
-    annoFile.foreach { annoPath =>
+    outAnno.foreach { annoPath =>
       val outputFile = new java.io.PrintWriter(annoPath)
       outputFile.write(JsonProtocol.serialize(annotations.filter(_ match {
         case _: DeletedAnnotation       => false
         case _: EmittedComponent        => false
         case _: EmittedAnnotation[_]    => false
         case _: FirrtlCircuitAnnotation => false
+        case _: OutAnnoAnnotation       => false
         case _ => true
       })))
       outputFile.close()
     }
   }
 
+  // TODO: Filter out blackbox dumping from this FIRRTL step, let CIRCT do it
+
   // Top Generation
-  def executeTop(): Seq[ExtModule] = {
-    val annos = new FirrtlStage().execute(
-      Array.empty,
-      annotations ++ Seq(
-        RunFirrtlTransformAnnotation(Dependency[ReParentCircuit]),
-        RunFirrtlTransformAnnotation(Dependency[RemoveUnusedModules])
-      ) ++
-        topAnnos
-    )
+  def executeTop(): Unit = {
+    val annos = new FirrtlStage().execute(Array.empty, annotations)
+
     annos.collectFirst { case FirrtlCircuitAnnotation(circuit) => circuit } match {
       case Some(circuit) =>
-        dump(circuit, annos, topFir, topAnnoOut)
-        circuit.modules.collect { case e: ExtModule => e }
+        dump(circuit, annos)
       case _ =>
         throw new Exception(s"executeTop failed while executing FIRRTL!\n")
     }
   }
 
   // Top and harness generation
   def executeTopAndHarness(): Unit = {
-    // Execute top and get list of ExtModules to avoid collisions
-    val topExtModules = executeTop()
-
-    // order is determined by DependencyAPIMigration
-    val harnessAnnos =
-      harnessDotfOut.map(BlackBoxResourceFileNameAnno).toSeq ++
-        harnessTop.map(ht => ModuleNameSuffixAnnotation(rootCircuitTarget, s"_in${ht}")) ++
-        synTop.map(st => ConvertToExtModAnnotation(rootCircuitTarget.module(st))) ++
-        Seq(
-          LinkExtModulesAnnotation(topExtModules),
-          RunFirrtlTransformAnnotation(Dependency[ConvertToExtMod]),
-          RunFirrtlTransformAnnotation(Dependency[RemoveUnusedModules]),
-          RunFirrtlTransformAnnotation(Dependency[AvoidExtModuleCollisions]),
-          RunFirrtlTransformAnnotation(Dependency[AddSuffixToModuleNames])
-        )
+    executeTop()
```

Comment:
Is there a need to have `executeTop` and the `TopAndHarness` version of the functions? In the past, we needed this split since we separated out the FIRRTL runs on separate modules. Now that new flow supports each module being in a different file this isn't needed. I would just have a single entrypoint (single execute function) and remove the naming around top and harness (since this distinction doesn't matter anymore).

Reply:
Since we are using `mems.conf` generated by the `SFC` for the `Chisel` -> `LowFIRRTL` -> `Verilog` case, I think we need both `executeTop` and `executeTopAndHarness` to generate both the `top.mems.conf` and `harness.mems.conf`. https://github.com/ucb-bar/barstools/blob/13e2bb92ab7360f0ce3b2ff4d27b3024736111fb/src/main/scala/barstools/tapeout/transforms/GenerateTopAndHarness.scala#L69 this code seems to guide the `SFC` to generate `harness.mems.conf` instead of `top.mems.conf`. Would there be a cleaner way to generate both `top.mems.conf` & `harness.mems.conf` without maintaining `executeTop` and `executeTopAndHarness`?